pyiceberg/io/pyarrow.py

```diff
@@ -1612,6 +1616,7 @@ def __init__(
         self._bound_row_filter = bind(table_metadata.schema(), row_filter, case_sensitive=case_sensitive)
         self._case_sensitive = case_sensitive
         self._limit = limit
+        self._downcast_ns_timestamp_to_us = Config().get_bool(DOWNCAST_NS_TIMESTAMP_TO_US_ON_WRITE)
```

Comment:
Do we need to check the format version for downcasting? (We have the table_metadata already, so we have access to it)